python/libraries/bimvee/importIitVicon.py

```diff
@@ -129,45 +136,31 @@ def importVicon(**kwargs):
-                    try:
-                        poseDict = outDict['data'][bodyId]['pose6q']
-                    except KeyError:
-                        print('KeyError exception.. Creating new key', bodyId)
-                        uniqueIds.append(bodyId)
-                        outDict['data'][bodyId] = {'pose6q': {'ts': [], 'point': [], 'rotation': []}}
-                        poseDict = outDict['data'][bodyId]['pose6q']
                 poseDict['ts'].append(ts)
                 poseDict['point'].append(point)
                 poseDict['rotation'].append(rotation)
-                if not separateBodiesAsChannels:
-                    poseDict['bodyId'].append(bodyId.encode('utf-8'))
+                poseDict['bodyId'].append(bodyId)
             line = file.readline()
 
     # converting lists of strings to numpy arrays of objects
-    if separateBodiesAsChannels:
-        for id in uniqueIds:
-            outDict['data'][id]['pose6q']['ts'] = np.array(outDict['data'][id]['pose6q']['ts'], dtype=np.float64)
-            outDict['data'][id]['pose6q']['point'] = np.array(outDict['data'][id]['pose6q']['point'], dtype=np.float64)
-            outDict['data'][id]['pose6q']['rotation'] = np.array(outDict['data'][id]['pose6q']['rotation'], dtype=np.float64)
-            if kwargs.get('zeroTime', kwargs.get('zeroTimestamps', True)):
-                zeroTimestampsForAChannel(outDict['data'][id])
-        if kwargs.get('zeroTime', kwargs.get('zeroTimestamps', True)):
-            rezeroTimestampsForImportedDicts(outDict)
-        outDict['info']['uniqueIds'] = uniqueIds
-    else:
-        poseDict['ts'] = np.array(poseDict['ts'], dtype=np.float64)
-        poseDict['point'] = np.array(poseDict['point'], dtype=np.float64)
-        poseDict['rotation'] = np.array(poseDict['rotation'], dtype=np.float64)
-        poseDict['bodyId'] = np.array(poseDict['bodyId'])
-        poseDict['uniqueIds'] = np.unique(poseDict['bodyId'])
-        if kwargs.get('separateMarkersFromSegments', False):
-            outDict['data']['vicon'] = separateMarkersFromSegments(poseDict)
-        else:            
-            outDict['data']['vicon'] = {'pose6q': poseDict}
-            
-        if kwargs.get('zeroTime', kwargs.get('zeroTimestamps', True)):
-            zeroTimestampsForAChannel(outDict['data']['vicon']) # TODO: Zeroing in the separated channel case
+    poseDict['ts'] = np.array(poseDict['ts'], dtype=np.float64)
+    poseDict['point'] = np.array(poseDict['point'], dtype=np.float64)
+    poseDict['rotation'] = np.array(poseDict['rotation'], dtype=np.float64)
+    poseDict['bodyId'] = np.array(poseDict['bodyId'], dtype=object)
+    if kwargs.get('zeroTime', kwargs.get('zeroTimestamps', True)):
+        zeroTimestampsForADataType(poseDict)
+    if kwargs.get('separateBodiesAsChannels', False):
+        outDict['info']['uniqueIds'] = np.unique(poseDict['bodyId'])
+        separatedBodies = splitByLabel(poseDict, 'bodyId')
```

Comment:
Does splitByLabel() also delete the 'bodyId' key-value pair inside poseDict?

Reply:
No - it leaves it